core/iwasm/common/wasm_runtime_common.c

```diff
@@ -4215,29 +4544,33 @@ wasm_runtime_invoke_c_api_native(WASMModuleInstanceCommon *module_inst,
                                  uint32 argc, uint32 *argv, bool with_env,
                                  void *wasm_c_api_env)
 {
-    wasm_val_t params_buf[16], results_buf[4];
+    wasm_val_t params_buf[16] = { 0 }, results_buf[4] = { 0 };
     wasm_val_t *params = params_buf, *results = results_buf;
     wasm_trap_t *trap = NULL;
     bool ret = false;
     wasm_val_vec_t params_vec, results_vec;
 
-    if (func_type->param_count > 16
-        && !(params = wasm_runtime_malloc(sizeof(wasm_val_t)
-                                          * func_type->param_count))) {
-        wasm_runtime_set_exception(module_inst, "allocate memory failed");
-        return false;
+    if (func_type->param_count > 16) {
+        if (!(params = wasm_runtime_malloc(sizeof(wasm_val_t)
+                                           * func_type->param_count))) {
+            wasm_runtime_set_exception(module_inst, "allocate memory failed");
+            return false;
+        }
+        memset(params, 0, sizeof(wasm_val_t) * func_type->param_count);
```

Comment:
Had better use 'runtime_malloc(sizefo(wasm_val_t) * func_type->param_count, module_inst, NULL, 0)' instead, we have defined runtime_malloc function in this file, it will memset the memory to 0, and throw exception when malloc fails.